.github/workflows/nightly.yml

```diff
@@ -0,0 +1,76 @@
+name: Nightly
+
+on:
+  schedule:
+    - cron: '0 2 * * *' # Daily at 02:00 runs only on default branch
+  workflow_dispatch:
+
+jobs:
+  nightly:
+    name: build / php-${{ matrix.php }}-${{ matrix.ts }}-${{ matrix.name }}-${{ matrix.compiler }}
+
+    runs-on: ${{ matrix.os }}
+    strategy:
+      fail-fast: false
+      matrix:
+        include:
+          - { name: linux, php: '8.2', ts: 'nts', compiler: 'gcc',   os: ubuntu-20.04 }
```

Comment:
The PHP 8.2 support should be added in a separate PR.
It should also be best done by me since I've some points to check regarding tests / skipping rules.
If everything is OK, I can make this PR within few days.

Reply:
This workflow can run builds and tests even if you don't add support for PHP 8.2.
The workflow runs daily at 02:00 AM and checks if the project can be built for PHP 8.2. Thus, it will help to understand what needs to be done before we can add PHP 8.2 to the main workflow. Nightly builds are standalone builds that don't affect main CI even if they fail.

Reply:
Let me explain main purpose of the night builds. Of course you can easily modify C guards and add new PHP version to main matrix. But in this case main workflow will fail every time till you fix everything.

But, with nightly builds you'll able to test your code changes against new PHP version (include linux and macos builds) without affecting main workflow. Futhermore, when you add code changes for the new PHP version you'll be able to have a guarranty that new changes doesn't break anything for currently supported versions and in this time to have a checks for a new version.

Other things that can be useful - it's periodical checks against pre-release PHP versions. You'll be able to see what kind of changes you'll need to add in futute and this is useful because whith this information you can planning release for new PHP version as soon as possible.... even, maybe, at the same time as PHP will be released :)

Reply:
Sorry for the delay, I was busy + hollydays.

> Other things that can be useful - it's periodical checks against pre-release PHP versions. You'll be able to see what kind of changes you'll need to add in futute and this is useful because whith this information you can planning release for new PHP version as soon as possible.... even, maybe, at the same time as PHP will be released :)

I think that you are hugely overestimating the cost of adding a new PHP version support for this kind of extension. This is not the kind of nitghmare it can be for a huge PHP library for instance.

When a new PHP version is available, I just create a branch, add/modify few lines, and I modify the main CI workflow in the same branch in order to target the new version in addition of the other supported versions.
Then, once I get the green flag from CI I merge it.

Why would I need a more complex process ? The main workflow you have added perfectly does the job.

The only thing I see that can be improved is the delay between the release date of the new PHP version and the moment I'm aware of it. But this is not a big issue for a yearly event.

Reply:
Okay, I understand your point of view. It's good that adding support for the new version is as easy as drinking a cup of coffee :)
I support your approach to keep CI/CD as simple as possible, so I'll close this PR.

Reply:
> Okay, I understand your point of view. It's good that adding support for the new version is as easy as drinking a cup of coffee :)

The most complex PHP version targeting is between 5.x & 7+, as you may guess, and it was implemented before the first release.